### PR TITLE
fix: Use a regex to filter files for processing

### LINKED ISSUE #48: Webpack and RSPack plugins require excluding font and other binary files
This is a follow up to #39 
Since bundlers deal with numerous file types using a allowed list instead of a blocked list seems like it would be a good idea.

My suggestion would be to use a regex that is override-able using config.

### PR DESCRIPTION
**Pull request by deepjoy**
fixes: #48

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export type StylexOptions = {
 
 export type UnpluginStylexOptions = {
   // compiler?: string
-  invalidExts?: string[]
+  validExts?: RegExp
```
Comment (eryue0220) [inline src/types.ts]: I think this should also support `string[]` or `string`, because only regexp maybe more complicate for some cases.

**Comment by eryue0220** [comment]
And we also need to handle the `invalidExts` options gracefully, because maybe someone has used this props to handle their business logic.

**Comment by eryue0220** [comment]
Thanks for your great work.

File: src/types.ts (hunk 1 above)
Comment (deepjoy) [reply src/types.ts]: added string[] that would allow a list of extensions like before.

**Comment by deepjoy** [comment]
> And we also need to handle the `invalidExts` options gracefully, because maybe someone has used this props to handle their business logic.

I don't think the invalidExts case makes sense. After adding to my repo I had to keep on adding every single extension as I added more files. 

Would prefer if the invalidExt were deprecated so that we don't have to handle edge cases like.
* User adds the same file extension to both invalidExt and validExt
* User adds a Regexp in validExt that also matches invalidExt

Since this is pre 1.0 software can this be a breaking change?

**Comment by deepjoy** [comment]
Also, I had to update esbuild and regenerate the lock file to fix the failing CI.

before the lock file update the build seemed quite unpredictable 
`ubuntu with node 18` [passed](https://github.com/eryue0220/unplugin-stylex/actions/runs/10003846012/job/27651440247) 
`ubuntu with node 16` [failed](https://github.com/eryue0220/unplugin-stylex/actions/runs/10003846012/job/27651439861)

**Comment by eryue0220** [comment]
> > And we also need to handle the `invalidExts` options gracefully, because maybe someone has used this props to handle their business logic.
> 
> I don't think the invalidExts case makes sense. After adding to my repo I had to keep on adding every single extension as I added more files.
> 
> Would prefer if the invalidExt were deprecated so that we don't have to handle edge cases like.
> 
> * User adds the same file extension to both invalidExt and validExt
> * User adds a Regexp in validExt that also matches invalidExt
> 
> Since this is pre 1.0 software can this be a breaking change?

@deepjoy 
 
Thank you for feedback.

Sure, you're right. We currently just in 0.4.x, breaking changes will be okay. But I still insist that we should ship the breaking change carefully, it has a bad DX. 

As for why I suggest that keeping the `invalidExt`, currently we just landed 0.4.0, and the next patch two version I want to fix some issues if anyone commits. Thus we can publish this feature the  in `0.4.1` or `0.4.2`.

In `0.5.0`, we'll remove the `invalidExt` completely.  

Or we could keep this pr currently, I'll plan to publish `0.5.0` in next two or three weeks.

WDYT?

**Comment by deepjoy** [comment]
Sure. How would you like to handle the edge cases if we keep both?

**Comment by deepjoy** [comment]
Took a stab at defining the behavior. See if this makes sense to you.

* if only invalidExts is specified keep previous behavior
* if both invalidExts and validExts are specified throw error
* if only validExts is specified use validExts options
* if neither invalidExts nor validExts is specified use new built-in regex

**Comment by eryue0220** [comment]
> Took a stab at defining the behavior. See if this makes sense to you.
> 
> * if only invalidExts is specified keep previous behavior
> * if both invalidExts and validExts are specified throw error
> * if only validExts is specified use validExts options
> * if neither invalidExts nor validExts is specified use new built-in regex

Cool, this is make sense. It's awesome.

**Comment by eryue0220** [comment]
Thank you for your work again.

**Comment by eryue0220** [comment]
https://www.npmjs.com/package/unplugin-stylex/v/0.5.0

**Pull request merged**